src/main.rs

```diff
@@ -72,20 +91,43 @@ enum CountParameter {
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     //Parse commandline
     let cli = Cli::parse();
-    let network: String = if let Some(chain_name) = cli.network {
+
+    let (network, dictionary_url) = if let Some(ref chain_name) = cli.network {
```

Comment:
Network is a flag for cli that accepts only one optional string. How will you get a tuple here for dictionary_url?